src/reformat_libyuv.c

```diff
@@ -50,6 +56,113 @@ unsigned int avifLibYUVVersion(void)
 #pragma clang diagnostic pop
 #endif
 
+static avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * rgb);
+
+avifResult avifImageRGBToYUVLibYUV(avifImage * image, const avifRGBImage * rgb)
```

Comment:
Currently there are inconsistencies with `avifChromaUpsampling`:

- the name is misleading as it does not only impact upsampling but also YUV to RGB conversion results, with and without upsampling (because of different conversion results with libyuv).
- comment mentions `ignored for 444` but this is wrong: `avifImageYUVToRGBLibYUV()` will bail out also for 444 on `BEST_QUALITY` for example.
- comment mentions `avifRGBImageSetDefaults() prefers quality over speed` but `avifRGBImageSetDefaults()` sets `AVIF_CHROMA_UPSAMPLING_AUTOMATIC` whose comment says `prefers libyuv, else uses BEST_QUALITY` so it actually prefers libyuv-grade speed over bilinear quality.

---

I added `avifChromaDownsampling` to be consistent with `avifChromaUpsampling` but I suggest we fix the problems mentioned above in a future change, especially the naming, even if it is an API breaking change.

> We may need to have a new option, for example, to select sharp YUV.

Good idea. It may be of interest to @maryla-uc.

Reply:
Yannis: I will review this pull request on Monday. Sorry about the delay.

I agree with your analysis. `avifChromaUpsampling` is really a parameter for the `avifImageYUVToRGB()` function and does not need to be stored in the `avifRGBImage` struct. Its meaning also became somewhat inconsistent when chroma upsampling with bilinear filtering was added to libyuv so that there were two impmenetations (libavif built-in and libyuv) of bilinear interpolation. We should look into how to add a `flags` parameter (a bitmask) to the `avifImageYUVToRGB()` function and define the bit flags appropriately.

Similarly `avifChromaDownsampling` is really a parameter of the `avifImageRGBToYUV()` function. An advantage of making it a member of the `avifRGBImage` struct is that existing `avifImageRGBToYUV()` calls don't need to be modified and just need to be recompiled. We can consider adding a `flags` parameter to the `avifImageRGBToYUV()` function now, or doing that at the same time as adding a `flags` parameter to the `avifImageYUVToRGB()` function.

Reply:
Unless there is still a lot of comments to make on this PR, I suggest we move `avifChromaUpsampling` and `avifChromaDownsampling` from `avifRGBImage` to `avifImageYUVToRGB()` and `avifImageRGBToYUV()` arguments in another change.

Alternatively, we can make the change for `avifChromaUpsampling` first, and then do the change for `avifChromaDownsampling` in this PR before merging it.

Reply:
Yannis: Your suggestion is fine by me.